sdv/constraints/tabular.py

```diff
@@ -301,23 +317,24 @@ def _fit(self, table_data):
                 The Table data.
         """
         if self._high_is_scalar is None:
-            self._high_is_scalar = self._high not in table_data.columns
+            self._high_is_scalar = not isinstance(self._high, list)
         if self._low_is_scalar is None:
-            self._low_is_scalar = self._low not in table_data.columns
+            self._low_is_scalar = not isinstance(self._low, list)
 
         if self._high_is_scalar and self._low_is_scalar:
             raise TypeError('`low` and `high` cannot be both scalars at the same time')
         elif self._low_is_scalar:
-            self.constraint_columns = (self._high,)
-            self._dtype = table_data[self._high].dtype
+            self.constraint_columns = tuple(self._high)
         elif self._high_is_scalar:
-            self.constraint_columns = (self._low,)
-            self._dtype = table_data[self._low].dtype
+            self.constraint_columns = tuple(self._low)
         else:
-            self._dtype = table_data[self._high].dtype
+            self.constraint_columns = tuple(self._low + self._high)
+            if len(self.constraint_columns) > 2:
+                raise ValueError('`low` and `high` cannot be more than one column.')
```

Comment:
If we no longer need to check if `self._high` and `self._low` are in `table_data.columns`, we can move these checks to the `__init__` method. That way we can also make sure `self.constraint_columns` is correctly set after `__init__` (right now it could be a nested tuple, which seems invalid).

Reply:
Also for the last error message, it would be more readable as '`low` and `high` cannot both be more than one column.`

Reply:
WRT the message, I would change it further to something like: `low and high can only be multiple columns if the other one is scalar`

Reply:
> If we no longer need to check if `self._high` and `self._low` are in `table_data.columns`, we can move these checks to the `__init__` method. That way we can also make sure `self.constraint_columns` is correctly set after `__init__` (right now it could be a nested tuple, which seems invalid).

Good catch @katxiao I think there is an additional problem here: It seems like the current implementation is assuming that if the value is not a list then it is a scalar, but that should not be the case: A scalar value could also be a column name, so I would rather say that the check this block of code needs to recover the `in table.columns` check somehow.